src/stateMachine.ts

```diff
@@ -107,6 +107,34 @@ export class StateMachine<
     });
   }
 
+  /**
+   * Generate a Mermaid StateDiagram of the current machine.
+   *
+   * Order your transitions so that the first and last are terminals
+   */
+  toMermaid( title?: string ) {
+    const diagram: string[] = [];
+    if (title) {
+      diagram.push("---");
+      diagram.push(`title: ${title}`);
+      diagram.push("---");
+    }
+    diagram.push("stateDiagram-v2");
+    diagram.push(`  [*] --> ${String(this.transitions[0].fromState)}`);
+
+    this.transitions.forEach(({ event, fromState, toState }) => {
+      const from = String(fromState);
+      const to = String(toState);
+      const evt = String(event);
+      diagram.push(`  ${from} --> ${to}: ${evt}`);
+    });
+
+    const last = this.transitions[this.transitions.length - 1];
```

Comment:
this is not necessarily a terminating state

Reply:
I agree. I am not sure how to create a better way. Maybe have the user pass in the terminals?

Maybe something like... `StateMachine#toMermaid("title", [States.closed, States.broken])`?

and in the generator, use the second array's [0] and [1] instead of the implicit order of the transitions?

Reply:
i'll fix it